runner/engine.go

```diff
@@ -655,8 +658,14 @@ func (e *Engine) stopBin() {
 			close(exitCode)
 		}
 	})
-	if ret := <-exitCode; ret != 0 {
-		os.Exit(ret)
+
+	select {
+	case ret := <-exitCode:
+		if ret != 0 {
+			e.exiter.Exit(ret) // Use exiter instead of direct os.Exit
```

Comment:
Add a comment about the fact it's for tests purpose

Reply:
done